pulser/simulation/simconfig.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2020 Pulser Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Union
+from typing_extensions import Literal, get_args
```

Comment:
Can you please do the conditional import that is in `sequence.py`? I would rather have users install `typing_extensions` only if they're on Python 3.7, it doesn't make sense to demand that from a user working on Python 3.8.